python/test_python.py

```diff
@@ -129,3 +129,55 @@ def test_write_file(self, docker_runner):
         docker_runner.run(['/bin/sh', '-c', 'python write_file.py output.txt "Bob Barker"; cat output.txt'])
         docker_runner.container.wait()
         assert str(docker_runner.container.logs(), 'UTF-8') == "BOB BARKER" # note no new line char
+
+
+class TestWriteJsonToStdout:
+    def test_json_array(self, docker_runner):
+        """Test that JSON array is parsed correctly"""
+        # Write string args as an array of strings to stdout
+        docker_runner.run('python json_stdout/array.py a b c d')
+        docker_runner.container.wait()
+        assert str(docker_runner.container.logs(), 'UTF-8') == '["a", "b", "c", "d"]\n'
+
+    def test_json_numbers(self, docker_runner):
+        """Test that JSON list of numbers is parsed correctly"""
+        # Write to stdout the length of each string argument
+        docker_runner.run('python json_stdout/numbers.py a bc def ghij')
+        docker_runner.container.wait()
+        print(docker_runner.container.logs())
+        assert str(docker_runner.container.logs(), 'UTF-8') == '[1, 2, 3, 4]\n'
```

Comment:
Let's parse the script's outputted JSON so we can compare them as Python arrays/dicts/strings/etc.
E.g, something like this:
```python
assert json.loads(str(docker_runner.container.logs(), 'UTF-8')) == [1, 2, 3, 4]
```

Also, at this point it might make sense to clean it up a bit so that the assertion can look like:
```python
assert script_output == [1, 2, 3, 4]
```

The benefits of parsing:
- it helps us test what we actually care about (e.g. the strings `'[1,2,3,4]'` and `'[ 1 , 2 , 3 , 4 ]'` are both good output but wouldn't both pass the test)
- the tests will be easier to read (e.g. `assert script_output == "hello 🥸"` vs `... == '"hello\\ud83e\\udd78"'`)

Reply:
great idea. That's so much cleaner and nicer to read in the tests. Also json.dumps takes bytes as well as strings so we can skip casting the bytes->str step